custom_components/thermal_comfort/config_flow.py

```diff
@@ -0,0 +1,146 @@
+    :returns: parameter value, or default value or None
+    """
+    if config_entry is not None:
+        return config_entry.options.get(param, config_entry.data.get(param, default))
+    else:
+        return default
+
+
+def build_schema(
+    config_entry: config_entries | None, show_advanced: bool = False
+) -> vol.Schema:
+    """Build configuration schema.
+
+    :param config_entry: config entry for getting current parameters on None
+    :param show_advanced: bool: should we show advanced options?
+    :return: Configuration schema with default parameters
+    """
+    # ToDo: get list of CONF_TEMPERATURE_SENSOR and CONF_HUMIDITY_SENSOR to create dropdown list and "one of" selection
+    schema = vol.Schema(
+        {
+            vol.Required(
+                CONF_NAME, default=get_value(config_entry, CONF_NAME, DEFAULT_NAME)
+            ): str,
+            vol.Required(
+                CONF_TEMPERATURE_SENSOR,
+                default=get_value(config_entry, CONF_TEMPERATURE_SENSOR),
+            ): str,
+            vol.Required(
+                CONF_HUMIDITY_SENSOR,
+                default=get_value(config_entry, CONF_HUMIDITY_SENSOR),
+            ): str,
+        }
+    )
+    if show_advanced:
+        schema = schema.extend(
+            {
+                vol.Optional(
+                    CONF_POLL, default=get_value(config_entry, CONF_POLL, False)
+                ): bool,
+            }
+        )
+        for st in SensorType:
+            default_enable = st in DEFAULT_SENSOR_TYPES
+            schema = schema.extend(
+                {
+                    vol.Optional(
+                        str(st),
+                        default=get_value(config_entry, str(st), default_enable),
+                    ): bool
+                }
+            )
+
+    return schema
+
+
+class ThermalComfortConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Configuration flow for setting up new thermal_comfort entry."""
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry):
+        """Get the options flow for this handler."""
+        return ThermalComfortOptionsFlow(config_entry)
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        errors = {}
+
+        if user_input is not None:
+            er = entity_registry.async_get(self.hass)
+
+            t_sensor = er.async_get(user_input[CONF_TEMPERATURE_SENSOR])
+            p_sensor = er.async_get(user_input[CONF_HUMIDITY_SENSOR])
+            _LOGGER.debug(f"Going to use t_sensor {t_sensor}")
+            _LOGGER.debug(f"Going to use p_sensor {p_sensor}")
+
+            if t_sensor is None:
+                self.async_abort(reason="Temperature sensor not found")
+
+            if p_sensor is None:
+                self.async_abort(reason="Pressure sensor not found")
+
+            await self.async_set_unique_id(f"{t_sensor.unique_id}-{p_sensor.unique_id}")
```

Comment:
This has to handle the case if the input sensors aren't in entity_registry == without unqiue_id. If one of them isn't found we shouldn't abort but set the unique_id to Config Entry ID.

Reply:
How `DeviceThermalComfort` will work if temperature or humidity sensor is missed?
Sensor not in registry == there is no such sensor. Isn't it?

Reply:
No the `entity_registry` is just for entities with unique_id's.
>The entity registry is a registry where Home Assistant keeps track of entities. Any entity that is added to Home Assistant which specifies the unique_id attribute will be registered in the registry.

You can test this with some simple template sensors as input
```yaml
template:
  - sensor:
    - name: Test Temperature
      state: 22.0
      device_class: 'temperature'
      unqiue_id: '324781092837410923874'
    - name: Test Humidity
      state: 50.0
      device_class: 'humidity'
```

Reply:
You could check in `hass.states` http://dev-docs.home-assistant.io/en/master/api/core.html#homeassistant.core.StateMachine.get

Reply:
> This has to handle the case if the input sensors aren't in entity_registry == without unqiue_id. If one of them isn't found we shouldn't abort but set the unique_id to [Config Entry ID](https://developers.home-assistant.io/docs/entity_registry_index/#unique-id-of-last-resort).

We have no `Config Entry ID` at this stage. It happens before.

Reply:
So check if unique_id is None in `async_setup_entry` and set it so `config_entry.entry_id` if it is?

Reply:
Not sure about this... Will check.

Reply:
```python
        if config_entry.unique_id is None:
            config_entry.unique_id = config_entry.entry_id
```

Reply:
I understand how it may be done. 
I'm not sure that this is legal trick and going to knew something new about this before making conclusions. 
For now (for me) it looks like `unique_id` must exist after pressing "Confirm" button in UI.

Reply:
https://github.com/home-assistant/core/pull/63547/files

in __init__ in setup_entry

Reply:
Hey! You did all interesting part of job ;)
Will move `unique_id` generation to `async_setup_entry`.